cpp/src/neighbors/detail/cagra/graph_core.cuh

```diff
@@ -1413,7 +1636,15 @@ void optimize(
     RAFT_LOG_DEBUG("# Pruning time: %.1lf sec", time_prune_end - time_prune_start);
   }
 
-  auto rev_graph       = raft::make_host_matrix<IdxT, int64_t>(graph_size, output_graph_degree);
+  // Allocate memory for rev_graph using Transparent HugePage
+  constexpr size_t thp_size = 2 * 1024 * 1024;
+  size_t byte_size          = sizeof(IdxT) * graph_size * output_graph_degree;
+  if (byte_size % thp_size) { byte_size += thp_size - (byte_size % thp_size); }
+  mmap_owner rev_graph(byte_size);
+  IdxT* rev_graph_ptr = (IdxT*)rev_graph.data();
+  memset(rev_graph_ptr, 0, byte_size);
```

Comment:
I think we can remove the memset here. We use the `MAP_ANONYMOUS`, which seems to zero-initialize the memory on all supported architectures https://stackoverflow.com/a/65084762 . I did a few benchmarks using this branch over the last week and found that the `memset` here can sometimes take excessive amounts of time when the graph is larger. Not sure why though. Perhaps, memory fragmentation?